src/traces/scattersmith/attributes.js

```diff
@@ -0,0 +1,70 @@
+'use strict';
+
+var hovertemplateAttrs = require('../../plots/template_attributes').hovertemplateAttrs;
+var texttemplateAttrs = require('../../plots/template_attributes').texttemplateAttrs;
+var extendFlat = require('../../lib/extend').extendFlat;
+var scatterAttrs = require('../scatter/attributes');
+var lineAttrs = scatterAttrs.line;
+
+module.exports = {
+    mode: scatterAttrs.mode,
+
+    re: {
+        valType: 'data_array',
+        editType: 'calc+clearAxisTypes',
+        description: 'Sets the real coordinates'
+    },
+
+    im: {
+        valType: 'data_array',
+        editType: 'calc+clearAxisTypes',
+        description: 'Sets the imaginary coordinates'
+    },
+
+    text: scatterAttrs.text,
+    texttemplate: texttemplateAttrs({editType: 'plot'}, {
+        keys: ['re', 'im', 'text']
+    }),
+    hovertext: scatterAttrs.hovertext,
+
+    line: {
+        color: lineAttrs.color,
+        width: lineAttrs.width,
+        dash: lineAttrs.dash,
+        shape: extendFlat({}, lineAttrs.shape, {
+            values: ['linear', 'spline']
+        }),
+        smoothing: lineAttrs.smoothing,
+        editType: 'calc'
+    },
+    connectgaps: scatterAttrs.connectgaps,
+
+    marker: scatterAttrs.marker,
+    cliponaxis: extendFlat({}, scatterAttrs.cliponaxis, {dflt: false}),
```

Comment:
Please test `cliponaxis: true` in the image test.

Reply:
For smith charts, there's not much meaning in plotting values outside the main plot area--that is, `cliponaxis` should always be `true` and this shouldn't be an attribute in the trace. Can I remove it?